osutil/syncdir.go

```diff
@@ -49,7 +58,7 @@ func (fref FileReference) State() (io.ReadCloser, int64, os.FileMode, error) {
 	if err != nil {
 		return nil, 0, os.FileMode(0), err
 	}
-	return file, fi.Size(), fi.Mode(), nil
+	return file, fi.Size(), fi.Mode().Perm(), nil
```

Comment:
Why these changes?

Reply:
This is needed to ensure that if ModeType was mistaknely passed to either `FileReferencePlusMode` and `MemoryFileState`, it wouldn't have effect (it already doesn't as AtomicWrite ignores mode type anyway).
This is needed to avoid `FileReferencePlusMode` and `MemoryFileState` passed with `os.ModeSymlink` messing this check [here](https://github.com/snapcore/snapd/pull/13071/files#diff-c96c5c4903c95e4dc4fa6b1e968c98ceb20c526551ec6509f6670cdc80489201R314).

There was a discussion about this with @pedronis https://github.com/snapcore/snapd/pull/13071#discussion_r1287066580.

Reply:
done